### PR TITLE
#1132 Add annotation and reflection sections to supported utilities section

### PR DESCRIPTION
**Pull request by hglasgow**
## Overview

Added two sections detailing annotation and reflection support, paraphrasing the javadoc.

---

I hereby agree to the terms of the JUnit Contributor License Agreement.

### CODE REVIEW

File: documentation/src/docs/asciidoc/extensions.adoc [hunk 1]
```diff
@@ -326,6 +326,71 @@ with annotations, reflection, and classpath scanning tasks. `TestEngine` and `Ex
 authors are encouraged to use these supported methods in order to align with the behavior
 of the JUnit Platform.
 
+==== Annotation Support
+
+`AnnotationSupport` provides a number of static utility methods that operate upon annotated
+(and meta-annotated) classes, as follows:
```
Comment (marcphilipp) [inline documentation/src/docs/asciidoc/extensions.adoc]: The also operate on all kinds of `AnnotatedElements`, e.g. methods etc.

**Comment by sbrannen** [comment]
To clarify: it is totally fine to provide links to the online JavaDoc for these supported utility classes, but that would be in addition to content suitable for the User Guide (as I described above).

**Comment by hglasgow** [comment]
Not a problem. I'll rework it in more of a prose style.

**Comment by sbrannen** [comment]
> Not a problem. I'll rework it in more of a prose style.

Thanks!

By the way, you don't need to document or provide examples for every single method in those utility classes. Rather, the point of the documentation is to inform engine and extension authors that the utilities exist and point them to the JavaDoc for further detailed information. 😉

**Comment by hglasgow** [comment]
Okay, that's updated. I'm happy to continue revising it if necessary.

**Comment by sbrannen** [comment]
> Okay, that's updated.

Thanks! That looks much better.

> I'm happy to continue revising it if necessary.

I'll go ahead and merge it _as is_ and perhaps perform some additional polishing after the merge.

**Comment by sbrannen** [comment]
This has been merged into `master`.

Cheers

**Pull request merged**